packages/relay/src/lib/precheck.ts

```diff
@@ -100,7 +100,7 @@ export class Precheck {
   chainId(tx: Transaction, requestId?: string) {
     const requestIdPrefix = formatRequestIdMessage(requestId);
     const txChainId = prepend0x(Number(tx.chainId).toString(16));
-    const passes = txChainId === this.chain;
+    const passes = txChainId === '0x0' || txChainId === this.chain;
```

Comment:
should we tighten this up a bit by checking to see if the v value is 27 or 28 (for legacy transactions) when the chain_id is zero?

Reply:
Yes this makes sense! Just pushed in an update for it